### PR TITLE
[op-conductor] Implement Raft FSM

### PR DESCRIPTION
**Pull request by 0x00101010**
**Description**

Implement Raft FSM that will be used by raft consensus protocol to store latest unsafe head payload for properly starting / stopping sequencer.

**Tests**

Added unit tests for the FSM.

**Additional context**

A follow up PR will be focused on the Raft implementation that utilizes this FSM.

**Metadata**

- https://github.com/ethereum-optimism/protocol-quest/issues/46

### CODE REVIEW

File: op-service/eth/ssz.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ type BlockVersion int
 
 const ( // iota is reset to 0
 	BlockV1 BlockVersion = iota
-	BlockV2              = iota
```
Comment (0x00101010) [inline op-service/eth/ssz.go:18]: With this syntax, BlockV2 is treated as int instead of `Blockversion`

File: op-conductor/consensus/raft_fsm.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+func (t *unsafeHeadTracker) Snapshot() (raft.FSMSnapshot, error) {
+	t.mtx.RLock()
+	defer t.mtx.RUnlock()
+
+	return &snapshot{
+		unsafeHead: t.unsafeHead,
+	}, nil
+}
+
+// CommitUnsafePayload commits the unsafe payload to the raft log.
```
Comment (zhwrd) [inline op-conductor/consensus/raft_fsm.go]: docstring typo

**Review by zhwrd — APPROVED** [review]

File: op-service/eth/ssz.go (hunk 1 above)
Comment (mslipper) [reply op-service/eth/ssz.go:18]: Just want to make sure - this does not change the semantics around serialization/deserialization, correct? It just makes the `BlockV2` constant a `BlockVersion`?

**Review by mslipper — APPROVED** [review]
Left one non-blocking nit assuming the `BlockVersion` semantics stay the same.

File: op-conductor/consensus/raft_fsm.go (hunk 2 above)
Comment (0x00101010) [reply op-conductor/consensus/raft_fsm.go]: Thanks for catching it, fixed

File: op-service/eth/ssz.go (hunk 1 above)
Comment (0x00101010) [reply op-service/eth/ssz.go:18]: Correct, and I double checked, 
1. MarshalSSZ does not serialize BlockVersion into data at all.
2. There're enough test cases where they compare BlockV2 as BlockVersion, so the change shouldn't affect that part as well

**Pull request merged**